lib/scraper_test.rb

```diff
@@ -30,17 +31,41 @@ def install_tasks
           it 'should contain the expected data' do
             # TODO: Make this configurable
             Dir['test/data/*.yml'].each do |file|
-              yaml_data = YAML.load_file(file).to_h
-              url = yaml_data[:url]
-              class_to_test = Object.const_get(yaml_data[:class])
-              VCR.use_cassette(File.basename(url)) do
-                response = class_to_test.new(response: Scraped::Request.new(url: url).response)
-                response.to_h.must_equal yaml_data[:to_h]
+              VCR.use_cassette(File.basename(file)) do
+                test = Instructions.new(file)
+                response = test.class_to_test.new(response: Scraped::Request.new(url: test.url).response)
+                response.to_h.must_equal test.expected
               end
             end
           end
         end
       end
     end
   end
+
+  class Instructions
+    def initialize(file)
+      @file = file
+    end
+
+    def url
+      yaml_data[:url] or raise "No url provided by #{File.basename(file)}."
+    end
+
+    def class_to_test
+      Object.const_get(yaml_data[:class]) or raise "No class provided by #{File.basename(file)}."
+    end
+
+    def expected
+      yaml_data[:to_h] or raise "No to_h provided by #{File.basename(file)}."
+    end
+
+    private
+
+    attr_reader :file
+
+    def yaml_data
```

Comment:
I don't think there's any value in coupling the method name here to the underlying data format. If we later allow people to supply the instructions in a JSON file, for example, we shouldn't need to go through and change all the places that call this method (or worse, access the JSON via a `yaml_data` method).